controllers/datafoundation.go

```diff
@@ -203,26 +203,22 @@ func (r *dataFoundationReconciler) reconcilePhases() (ctrl.Result, error) {
 func (r *dataFoundationReconciler) parseSpec(offering *v1alpha1.ManagedFusionOffering) error {
 	r.Log.Info("Parsing ManagedFusionOffering Data Foundation spec")
 
+	var dfSpec dataFoundationSpec
+	err := yaml.Unmarshal([]byte(offering.Spec.Config), &dfSpec)
+	if err != nil {
+		r.Log.Error(err, "an error occurred while unmarshaling the YAML config")
+		return fmt.Errorf("failed to unmarshal YAML config")
+	}
 	isValid := true
-	var err error
-	var usableCapacityInTiB int
-	usableCapacityInTiBAsString, found := offering.Spec.Config["usableCapacityInTiB"]
-	if !found {
+	if dfSpec.usableCapacityInTiB == 0 {
```

Comment:
I am a bit sceptical about this. any suggestions?

Reply:
0 is the default value, right? 
If nothing is passed then 0 will be set, right?